jquery.flot.gldrawSeries.js

```diff
@@ -0,0 +1,154 @@
+         * @param {*} series 
+         * @param {*} scene 
+         * @param {*} plotOffset 
+         * @param {*} plotWidth 
+         * @param {*} plotHeight 
+         * @param {*} drawSymbol 
+         * @param {*} getColorOrGradient 
+         */      
+        function drawSeriesPoints(series, plotscene, mainscene, plotOffset, plotWidth, plotHeight, drawSymbol, getColorOrGradient) {
+            var texture = plotscene.userData.texture;
+            var material = plotscene.userData.material;
+            var geometry = plotscene.userData.geometry;
+            var rendererSize = mainscene.userData.rendererSize;
+            mainscene.userData.plotOffset = plotOffset;
+
+            // create point texture
+            if (!texture) {
+                var symbol = series.points.symbol;
+                var textureCanvas = document.createElement('canvas');
+                textureCanvas.width = 128;
+                textureCanvas.height = 128;
+                var context = textureCanvas.getContext('2d');
+                context.clearRect(0, 0, 128, 128);
+                context.globalAlpha = 1;
+                context.strokeStyle = series.color;
+                context.lineWidth = series.points.lineWidth * 10 || 10;
+
+                if (symbol === "circle") {
+                    context.arc(64, 64, 32, 0, 2 * Math.PI);
+                } else if (typeof symbol === 'string' && drawSymbol && drawSymbol[symbol]) {
+                    drawSymbol[symbol](context, 64, 64, 32, false);
+                } else if (typeof drawSymbol === 'function') {
+                    drawSymbol(context, 64, 64, 32, false);
+                }
+
+                context.fillStyle = getFillStyle(series.points, series.color, null, null, getColorOrGradient) || series.color;
+                context.fill();
+                context.stroke();
+                texture = new THREE.Texture(textureCanvas);
+                texture.needsUpdate = true;
+
+                // save texture for future draw
+                plotscene.userData.texture = texture;
+            }
+
+            // update points material from texture
+            if (!material || texture.needsUpdate) {
+                material = new THREE.PointsMaterial(
+                    {
+                        size: series.points.radius * 4,
+                        sizeAttenuation: false,
+                        map: texture,
+                        transparent: true,
+                        alphaTest: 0.1
+                    });
+                material.needsUpdate = true;
+                // save material for future draw
+                plotscene.userData.material = material;
+            }
+
+            if (!geometry) {
+                geometry = plotscene.userData.geometry = new THREE.Geometry();
+            } 
+
+            // clear the scene
+            while (plotscene.children.length > 0) {
+                plotscene.remove(plotscene.children[0]);
+            }
```

Comment:
plotscene is a THREE.Scene object for now, but should be change to a simple object. Currently all Points are added to mainscene and plotscene is used only for saving plot texture, material and geometry.

Reply:
plotscene.children.length should be 0 because the points are added to mainscene, so i have to delete this :)